src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12773,9 +12917,21 @@ void Assembler::prefix(Address adr) {
       prefix(REX_X);
     }
   }
+  if (is_map1) emit_int8(0x0F);
 }
 
-void Assembler::prefix(Address adr, Register reg, bool byteinst) {
+void Assembler::prefix_rex2(Address adr, bool is_map1) {
+  int bits = is_map1 ? REX2BIT_M0 : 0;
+  bits |= get_base_prefix_bits(adr.base()->encoding());
```

Comment:
```suggestion
  bits |= get_base_prefix_bits(adr.base());
```

Reply:
As per section 3.7.5 of Intel SDM (Index ∗ Scale) + Displacement is a valid addressing mode.  Thus we should set the bits corresponding to extended base register encoding only if its a valid register.

Reply:
Thank you @jatin-bhateja.  I've made the change.